src/Provider/HostIp/AbstractHostIp.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\HostIp;
+
+use Geocoder\Collection;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+
+/**
+ * @author William Durand <william.durand1@gmail.com>
+ * @author Oleg Andreyev <oleg@andreyev.lv>
+ */
+abstract class AbstractHostIp extends AbstractHttpProvider
```

Comment:
The class needs to implement `Geocoder\Provider\Provider`

Reply:
it's already implemented by \Geocoder\Provider\AbstractProvider

`AbstractHttpProvider -> AbstractProvider -> Provider`

Reply:
I do know that ; this is just for consistency !

Reply:
Indeed ; though, could you add it for consistency with the other providers ?

Reply:
ok.